org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/GradleClasspathContainerUpdater.java

```
@@ -118,42 +117,27 @@ public IClasspathEntry apply(OmniExternalDependency dependency) {
-     * @param monitor                the monitor to report progress on
-     * @throws JavaModelException if the container assignment fails
      */
-    public static void updateFromModel(IJavaProject eclipseProject, OmniEclipseProject gradleProject, Set<OmniEclipseProject> allGradleProjects, IProgressMonitor monitor) throws JavaModelException {
+    public static void updateFromModel(IJavaProject eclipseProject, OmniEclipseProject gradleProject, Set<OmniEclipseProject> allGradleProjects, PersistentModel model, PersistentModelBuilder updates, IProgressMonitor monitor) throws JavaModelException {
         GradleClasspathContainerUpdater updater = new GradleClasspathContainerUpdater(eclipseProject, gradleProject, allGradleProjects);
-        updater.updateClasspathContainer(monitor);
+        updater.updateClasspathContainer(model, updates, monitor);
     }
 
     /**
      * Updates the classpath container from the state stored by the last call to {@link #updateFromModel(IJavaProject, OmniEclipseProject, IProgressMonitor)}.
-     *
-     * @param eclipseProject the target project to update the classpath container on
-     * @param monitor the monitor to report progress on
-     * @return true if the container could be loaded, false if the container remains uninitialized
-     * @throws JavaModelException if the classpath cannot be assigned
      */
     public static boolean updateFromStorage(IJavaProject eclipseProject, IProgressMonitor monitor) throws JavaModelException {
-        List<IClasspathEntry> storedClasspath = CorePlugin.modelPersistence().loadModel(eclipseProject.getProject()).getClasspath();
-        if (storedClasspath != null) {
-            setClasspathContainer(eclipseProject, storedClasspath, monitor);
-            return true;
-        } else {
+        PersistentModel model = CorePlugin.modelPersistence().loadModel(eclipseProject.getProject());
+        if (model.isEmptyModel()) {
```

We should just use null here. Alternatively you can let the empty model throw exceptions on all getters. 

But we cannot let the empty model just silently return defaults, as that would hide programming errors.